x-pack/plugins/cases/public/components/user_actions/comment/registered_attachments.tsx

```diff
@@ -41,11 +44,13 @@ type BuilderArgs<C, R> = Pick<
  * Provides a render function for attachment type
  */
 const getAttachmentRenderer = memoize((attachmentType: AttachmentType<unknown>) => {
-  const attachmentViewObject = attachmentType.getAttachmentViewObject();
-
   let AttachmentElement: React.ReactElement;
 
   const renderCallback = (props: object) => {
+    const attachmentViewObject = attachmentType.getAttachmentViewObject(
+      props as ExternalReferenceAttachmentViewProps
```

Comment:
In combination with the comment here https://github.com/elastic/kibana/pull/153957/files#r1154257681 we should remove the casting.

Reply:
nit: Also to avoid the extra call to `getAttachmentViewObject` maybe we can pass the `attachmentViewObject` as an argument to `renderCallback`.